### PR TITLE
integrated Program Functionality

### LINKED ISSUE #84: (UI) Add Programs with static list page for navigation
**Acceptance criteria:**

1) User should be able to see a static program list page

2) On click of "+" user should be navigated to "Add Program" page as referred in image 2.

3) The current available Donors & Supplement needs to be binded to the dropdowns(If value is not there we can insert records as per the python model for those objects

4) Once the user fills in all the details on click of save the details should be stored in programs collection

5) Once saved the user should be simply navigated back to Program list page

6) When user clicks cancel the user should be simply navigated back to Program list page

**Reference mocks:**

1) Programs list page 

Image

2) Add Program page 

Image

### PR DESCRIPTION
**Pull request by JaswinderSinghdigwa**
- integrated all functionality and UI in Program

### CODE REVIEW

File: Backend/config/database.py [hunk 1]
```diff
@@ -3,7 +3,7 @@
 MongoClient()
 
 client = MongoClient(
-    "mongodb+srv://username:password@cluster0.cythueh.mongodb.net/?retryWrites=true&w=majority")
+    "mongodb+srv://findjassi121212:Change456@cluster0.qzwhu7z.mongodb.net/?retryWrites=true&w=majority")
```
Comment (mjayakumarmca) [inline Backend/config/database.py]: Please revert this change

File: Backend/routes/routes.py [hunk 2]
```diff
@@ -213,7 +213,7 @@ async def delete_child(id: str):
     return {"status": "ok", "data": []}
 
 
-@child_malnutrition.post("/api/childMalnutrion_Add")
+@child_malnutrition.post("/childMalnutrion_Add")
```
Comment (mjayakumarmca) [inline Backend/routes/routes.py]: revert this change too. we need to have api

File: FrontEnd/http-common.js [hunk 3]
```diff
@@ -1,7 +1,7 @@
 import axios from "axios";
 
 export const HTTP = axios.create({
-  baseURL: `http://127.0.0.1:7000/api/`,
+  baseURL: `http://127.0.0.1:8000/`,
```
Comment (mjayakumarmca) [inline FrontEnd/http-common.js]: revert this change too

File: FrontEnd/src/router/index.js [hunk 4]
```diff
@@ -62,34 +62,19 @@ const router = createRouter({
       component: () => import("../views/EditAanganwadi.vue")
     },
     {
-      path: "/addSupplement",
```
Comment (mjayakumarmca) [inline FrontEnd/src/router/index.js]: you have delete all these changes which are required while rebasing. Can you bring back all. You should rebase carefully. You can take my or someone's help

File: FrontEnd/src/stores/program.js [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+  }),
+  getters: {
+    getProgram(state) {
+      return state.programs;
+    },
+  },
+  actions: {
+    async fetchProgram() {
+      try {
+        const data = await HTTP.get("api/get_programs");
```
Comment (mjayakumarmca) [inline FrontEnd/src/stores/program.js]: we can remove api/ here as it will be available in http-common file itself. so you can remove the api in all the below endpoint calls also.

File: FrontEnd/src/stores/program.js [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+      } catch (error) {
+        alert(error);
+        console.log(error);
+      }
+    },
+    async fetchSupplement() {
+      try {
+         const response = await HTTP.get("api/get_supplements_details")
+        this.supplement = response.data;
+        console.log("333",this.supplement)
```
Comment (mjayakumarmca) [inline FrontEnd/src/stores/program.js]: you can give this console comment with a meaningful name

File: FrontEnd/src/stores/program.js [hunk 7]
```diff
@@ -0,0 +1,92 @@
...
+
+      } catch (error) {
+        alert(error);
+        console.log(error);
+      }
+    },
+    async fecthDonor() {
+      try {
+        await HTTP.get("api/getdonors").then((response) => {
+          console.log("331",response.data)
```
Comment (mjayakumarmca) [inline FrontEnd/src/stores/program.js]: here too

File: FrontEnd/src/views/EditProgram.vue [hunk 8]
```diff
@@ -0,0 +1,131 @@
...
+
+const store = useProgramStore();
+const route = useRoute();
+
+let supplements = computed(() => {
+	return store.supplement
+})
+
+let donors = computed(() => {
+	return store.donor.data
```
Comment (mjayakumarmca) [inline FrontEnd/src/views/EditProgram.vue]: please correct this alignment and above one

File: FrontEnd/src/views/EditProgram.vue [hunk 9]
```diff
@@ -0,0 +1,131 @@
...
+  from_date:"",
+  to_date:""
+});
+
+currentProgram = computed(() => {
+  if (store.currentprogram) {
+    return store.currentprogram;
+  } else {
+    return {
+		title:"",
```
Comment (mjayakumarmca) [inline FrontEnd/src/views/EditProgram.vue]: here too

File: FrontEnd/src/views/EditProgram.vue [hunk 10]
```diff
@@ -0,0 +1,131 @@
...
+
+const updatingProgram = async () => {	
+  console.log("cureent",currentProgram.value)
+  await store.updatingProgram(currentProgram);
+  return router.push("/programlist");
+};
+</script>
+
+<template>
+	<div class="container-full mt-5 p-3">
```
Comment (mjayakumarmca) [inline FrontEnd/src/views/EditProgram.vue]: can you fix all these alignments

File: FrontEnd/src/views/kidsCheckUp.vue [hunk 11]
```diff
@@ -1,6 +1,6 @@
 <template>
-  <div class="full-div container" style="width: 1280px">
-    <div id="x-contest" class="container-fluid p-3">
+<main class="container-full">
```
Comment (mjayakumarmca) [inline FrontEnd/src/views/kidsCheckUp.vue]: will these changes work in small mobile view correctly?

File: FrontEnd/src/views/programlist.vue [hunk 12]
```diff
@@ -0,0 +1,113 @@
+<script setup>
+import { onMounted, computed } from "vue";
+import swal from "sweetalert";
+import { useProgramStore } from "../stores/program.js";
+import router from "../router";
+
+  const store = useProgramStore();
```
Comment (mjayakumarmca) [inline FrontEnd/src/views/programlist.vue]: alignment issue

File: main.py [hunk 13]
```diff
@@ -44,7 +44,7 @@
 
 app.mount("/static", StaticFiles(directory="static"), name="static")
 
-app.include_router(sign_router)
+# app.include_router(sign_router)
```
Comment (mjayakumarmca) [inline main.py]: if these changes are not needed can you revert those.

File: static/index.html [hunk 14]
```diff
@@ -31,4 +28,6 @@
 	<script src="https://unpkg.com/leaflet@1.9.3/dist/leaflet.js" defer></script>
 	
   </body>
+<style>
+</style>
```
Comment (mjayakumarmca) [inline static/index.html]: do we need this empty one?

**Review by mjayakumarmca — CHANGES_REQUESTED** [review]
Given few comments. Once you did the changes test it fully. Also once PR is raised do a self review for all the basics.

File: FrontEnd/src/views/kidsCheckUp.vue (hunk 11 above)
Comment (mjayakumarmca) [reply FrontEnd/src/views/kidsCheckUp.vue]: When I tested in my machine without these changes it looks fine for me. So you can revert the css changes in this file

**Comment by JaswinderSinghdigwa** [comment]
- updated the changes u want

**Comment by JaswinderSinghdigwa** [comment]
- Updated the changes

File: FrontEnd/src/stores/program.js [hunk 15]
```diff
@@ -0,0 +1,89 @@
...
+      try {
+        await HTTP.get("get_program/" + id).then((response) => {
+          this.currentprogram = response.data.data[0];
+        });
+      } catch (error) {
+        alert(error);
+        console.log(error);
+      }
+    },
+    async fetchSupplement() {
```
Comment (mjayakumarmca) [inline FrontEnd/src/stores/program.js:68]: Ideally we should have this in supplement store but merging for now for the demo

File: FrontEnd/src/stores/program.js [hunk 16]
```diff
@@ -0,0 +1,89 @@
...
+      try {
+        const response = await HTTP.get("get_supplements_details");
+        this.supplement = response.data;
+        console.log("333", this.supplement);
+      } catch (error) {
+        alert(error);
+        console.log(error);
+      }
+    },
+    async fecthDonor() {
```
Comment (mjayakumarmca) [inline FrontEnd/src/stores/program.js:78]: same here(Ideally we should have this in donors store but merging for now for the demo)

File: FrontEnd/src/views/kidsCheckUp.vue [hunk 17]
```diff
@@ -1,6 +1,6 @@
 <template>
-  <div class="full-div container" style="width: 1280px">
-    <div id="x-contest" class="container-fluid p-3">
+<div class="full-div container" style="width: 1280px">
+<div id='x-main' class="container-fluid mt-5 p-3">
```
Comment (mjayakumarmca) [inline FrontEnd/src/views/kidsCheckUp.vue]: this is not needed can you revert it back to original

File: main.py [hunk 18]
```diff
@@ -19,7 +19,6 @@
 from Backend.routes.routes import donor_router, supp_router, program_router
 
 
-from app.routes import sign_router
```
Comment (mjayakumarmca) [inline main.py]: why this is deleted can you revert (or) add it back

**Review by mjayakumarmca — APPROVED** [review]
LGTM
